### PR TITLE
expression: implement vectorized evaluation for `builtinJSONAnyValueSig`

### PR DESCRIPTION
**Pull request by Baytwt**
### What problem does this PR solve? 

some format unification, and 
expression: implement vectorized evaluation for `builtinJSONAnyValueSig`

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: expression/builtin_miscellaneous_vec.go [hunk 1]
```diff
@@ -26,6 +26,10 @@ import (
 	"github.com/pingcap/tidb/util/chunk"
 )
 
+func (b *builtinInetNtoaSig) vectorized() bool {
+	return true
+}
+
```
Comment (qw4990) [inline expression/builtin_miscellaneous_vec.go]: These changes seem unnecessary, please fix them.

File: expression/builtin_miscellaneous_vec.go [hunk 2]
```diff
@@ -58,7 +62,7 @@ func (b *builtinInetNtoaSig) vecEvalString(input *chunk.Chunk, result *chunk.Col
 	return nil
 }
 
-func (b *builtinInetNtoaSig) vectorized() bool {
+func (b *builtinIsIPv4Sig) vectorized() bool {
```
Comment (qw4990) [inline expression/builtin_miscellaneous_vec.go]: ditto

File: expression/builtin_miscellaneous_vec.go [hunk 3]
```diff
@@ -88,9 +92,6 @@ func (b *builtinIsIPv4Sig) vecEvalInt(input *chunk.Chunk, result *chunk.Column)
 	return nil
 }
 
-func (b *builtinIsIPv4Sig) vectorized() bool {
-	return true
-}
```
Comment (qw4990) [inline expression/builtin_miscellaneous_vec.go]: ditto

File: expression/builtin_miscellaneous_vec_test.go [hunk 4]
```diff
@@ -62,6 +62,8 @@ var vecBuiltinMiscellaneousCases = map[string][]vecExprBenchCase{
 		{retEvalType: types.ETDuration, childrenTypes: []types.EvalType{types.ETString, types.ETDuration}},
 		{retEvalType: types.ETString, childrenTypes: []types.EvalType{types.ETString, types.ETString}},
 		{retEvalType: types.ETInt, childrenTypes: []types.EvalType{types.ETString, types.ETInt}},
+		{retEvalType: types.ETJSON, childrenTypes: []types.EvalType{types.ETString, types.ETJSON}},
+		{retEvalType: types.ETDecimal, childrenTypes: []types.EvalType{types.ETString, types.ETDecimal}},
```
Comment (SunRunAway) [inline expression/builtin_miscellaneous_vec_test.go]: Why adding Decimal to `NameConst`? Maybe to`AnyValue`.

File: expression/builtin_miscellaneous_vec.go [hunk 5]
```diff
@@ -159,11 +160,11 @@ func (b *builtinNameConstStringSig) vecEvalString(input *chunk.Chunk, result *ch
 }
 
 func (b *builtinDecimalAnyValueSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinDecimalAnyValueSig) vecEvalDecimal(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	return b.args[1].VecEvalDecimal(b.ctx, input, result)
```
Comment (SunRunAway) [inline expression/builtin_miscellaneous_vec.go]: Is this modification related to this PR, or should we add a test case for this function signature?

**Comment by qw4990** [comment]
Hi @Baytwt , please resolve these CI problems and conflicts.

**Pull request closed without merging**